runtime/compiler/optimizer/J9Inliner.cpp

```diff
@@ -948,16 +950,36 @@ bool TR_J9InterfaceCallSite::findCallSiteTargetImpl(
                      }
                   }
 
+               comp()->exitHeuristicRegion();
+
                if (topProfiledClass != NULL
                    && remainingTotalFreq >= 32
                    && topProfiledFreq == remainingTotalFreq)
                   {
-                  testType = TR_VftTest;
-                  thisClass = topProfiledClass;
+                  if (!comp()->compileRelocatableCode()
+                      || (comp()->getOption(TR_UseSymbolValidationManager)
+                          && comp()
+                             ->getSymbolValidationManager()
+                             ->addProfiledClassRecord(topProfiledClass)
+                          && comp()
+                             ->getSymbolValidationManager()
+                             ->addClassInstanceOfClassRecord(topProfiledClass, iface, true, true, true)))
+                     {
+                     testType = TR_VftTest;
+                     thisClass = topProfiledClass;
+                     }
+                  else
+                     {
+                     calleeResolvedMethod = NULL;
```

Comment:
I don't think failure to create the records should require us to discard this call target. We can still inline it with a method test like we would have if we hadn't looked at the profiling